services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedCredentialsService.java

```diff
@@ -382,6 +383,8 @@ private CompositeFuture createIndices() {
                         .flatMap(ok -> Optional
                                 .ofNullable(result.getJsonArray(MongoDbDeviceRegistryUtils.FIELD_CREDENTIALS))
                                 .map(credential -> credential.getJsonObject(0))
+                                .filter(credential -> DeviceRegistryUtils.matchesWithClientContext(credential,
```

Comment:
if the result is a JsonArray (as the previous line seems to indicate), then we should probably try to find the matching credentials among the elements of that array instead of just matching the first element, right?

Reply:
AFAIK, at any case there won't be more than one element. The result is obtained by matching _tenantId_, _authId_ and _type_. No two records can have same value for those fields.

Reply:
Yes, that is true for our implementation at hand.